homeassistant/components/wiz/discovery.py

```diff
@@ -0,0 +1,61 @@
+"""The wiz integration discovery."""
+from __future__ import annotations
+
+import asyncio
+from dataclasses import asdict
+import logging
+
+from pywizlight.discovery import DiscoveredBulb, find_wizlights
+
+from homeassistant import config_entries
+from homeassistant.components import network
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_discover_devices(
+    hass: HomeAssistant, timeout: int, address: str | None = None
+) -> list[DiscoveredBulb]:
+    """Discover wiz devices."""
+    if address:
+        targets = [address]
+    else:
+        targets = [
+            str(address)
+            for address in await network.async_get_ipv4_broadcast_addresses(hass)
+        ]
+
+    combined_discoveries: dict[str, DiscoveredBulb] = {}
+    for idx, discovered in enumerate(
+        await asyncio.gather(
+            *[find_wizlights(timeout, address) for address in targets],
+            return_exceptions=True,
+        )
+    ):
+        if isinstance(discovered, Exception):
+            _LOGGER.debug("Scanning %s failed with error: %s", targets[idx], discovered)
+            continue
+        for device in discovered:
+            assert isinstance(device, DiscoveredBulb)
+            combined_discoveries[device.ip_address] = device
+
+    return list(combined_discoveries.values())
+
+
+@callback
+def async_trigger_discovery(
+    hass: HomeAssistant,
+    discovered_devices: list[DiscoveredBulb],
+) -> None:
+    """Trigger config flows for discovered devices."""
+    for device in discovered_devices:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_DISCOVERY},
```

Comment:
The source is generally the domain of the integration that found it. `SOURCE_DISCOVERY` was the source for the removed `discovery` integration.

Reply:
I've used `SOURCE_DISCOVERY` in these cases because otherwise it won't be in `DISCOVERY_SOURCES`

https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L110

If there is a better way, happy to adjust in the next turn

Reply:
So the thing is, it is to know how it was discovered, but this isn't discovered by the `discovery` integration.

My suggestion would be, to add a `SOURCE_INTEGRATION` or `SOURCE_CUSTOM` to indicate a custom discovery method provided by the integration.

That way, we can still know who/what discovered it.

Reply:
`SOURCE_INTEGRATION ` makes sense to me

Reply:
```
homeassistant/components/ezviz/camera.py:                    context={"source": SOURCE_DISCOVERY},
homeassistant/components/flux_led/discovery.py:                context={"source": config_entries.SOURCE_DISCOVERY},
homeassistant/components/senseme/discovery.py:                    context={"source": config_entries.SOURCE_DISCOVERY},
homeassistant/components/steamist/discovery.py:                context={"source": config_entries.SOURCE_DISCOVERY},
homeassistant/components/tplink/__init__.py:                context={"source": config_entries.SOURCE_DISCOVERY},
homeassistant/components/unifiprotect/discovery.py:                    context={"source": config_entries.SOURCE_DISCOVERY},
```

I'll do some refactoring

Reply:
Looks like we already have

`SOURCE_INTEGRATION_DISCOVERY = "integration_discovery"` but its missing from `DISCOVERY_SOURCES`

Reply:
Oh, would you look at that! 👍  Two whole integration are using it :)

Reply:
https://github.com/home-assistant/core/pull/65811
https://github.com/home-assistant/frontend/pull/11564